packages/@react-spectrum/label/src/StaticField.tsx

```diff
@@ -0,0 +1,180 @@
+import {useStyleProps} from '@react-spectrum/utils';
+
+interface StaticFieldBaseProps extends DOMProps, StyleProps, Omit<SpectrumLabelableProps, 'necessityIndicator' | 'isRequired'>, AriaLabelingProps {}
+
+type NumberValue = number | RangeValue<number>;
+interface NumberProps<T extends NumberValue> {
+    value: T,
+    formatOptions?: Intl.NumberFormatOptions
+}
+
+type DateTime = Date | CalendarDate | CalendarDateTime | ZonedDateTime | Time;
+type RangeDateTime = RangeValue<DateTime>;
+type DateTimeValue = DateTime | RangeDateTime;
+interface DateProps<T extends DateTimeValue> {
+    value: T,
+    formatOptions?: Intl.DateTimeFormatOptions
+}
+
+interface StringProps<T extends string> {
+    value: T
+}
+
+interface StringListProps<T extends string[]> {
+    value: T,
+    formatOptions?: Intl.ListFormatOptions
+}
+
+type StaticFieldProps<T> =
+    T extends NumberValue ? NumberProps<T> :
+    T extends DateTimeValue ? DateProps<T> :
+    T extends string[] ? StringListProps<T> :
+    T extends string ? StringProps<T> :
+    never;
+
+type SpectrumStaticFieldProps<T> = StaticFieldProps<T> & StaticFieldBaseProps;
+
+function StaticField<T>(props: SpectrumStaticFieldProps<T>, ref: RefObject<HTMLElement>) {
+  let {
+    value, 
+    formatOptions,
+
+    ...otherProps
+  } = props;
+  let {styleProps} = useStyleProps(otherProps);
+
+  return (
+    <div
+      {...styleProps}
+      ref={ref as RefObject<HTMLDivElement>}>
+      <Label
+        elementType="span">
+        <span>
+          {Array.isArray(props.value) && 
+            <FormattedStringList value={value as string[]} formatOptions={formatOptions as Intl.ListFormatOptions} />}
+
+          {typeof value === 'object' && 'start' in (value as any) && typeof (value as RangeValue<NumberValue>).start === 'number' && 
+            <FormattedNumber value={value as NumberValue} formatOptions={formatOptions as Intl.NumberFormatOptions}  />}
+
+          {typeof value === 'object' && 'start' in (value as any) && typeof (value as RangeValue<DateTimeValue>).start !== 'number' && 
+            <FormattedDateRange value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'number' && 
+            <FormattedNumber value={value as NumberValue} formatOptions={formatOptions as Intl.NumberFormatOptions} />}
+
+          {typeof value === 'object' && 'calendar' in (value as any) && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {value instanceof Date && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+          
+          {value instanceof Time && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'string' && 
+            props.value}
```

Comment:
I know you are still working on some of the typescript at the moment, but I'm curious if we need all the explicit value casting here (e.g. `value as string[]`). I figured the conditionals would help typescript infer the true type of value. I think it will work after adding the extends stuff we talked about to StaticField<T>

Reply:
I thought this too, but i haven't been able to get it to behave yet

Reply:
okay so i don't think i can get ride of things like value={value as string[]} without a typescript error but it is possible now to do `'calendar' in value` instead of `'calendar' in (value as any)` now